src/item.cpp

```diff
@@ -3845,10 +3846,11 @@ units::mass item::weight( bool include_contents, bool integral ) const
     }
 
     units::mass ret;
-    if( integral ) {
-        ret = units::from_gram( get_var( "integral_weight", to_gram( type->integral_weight ) ) );
+    std::string local_str_mass = integral ? get_var( "integral_weight" ) : get_var( "weight" );
+    if( local_str_mass.empty() ) {
+        ret = integral ? type->integral_weight : type->weight;
     } else {
-        ret = units::from_gram( get_var( "weight", to_gram( type->weight ) ) );
+        ret = units::from_milligram( _atoi64( local_str_mass.c_str() ) );
```

Comment:
Actually, I think you should use `has_var()` to check if the variables exist and use `get_var( "...", 0 )` to get the variable as an integer, so you can avoid using `_atoi64` here, which is not a standard function.

Reply:
going to use std::stoll